### PR TITLE
Add short description field to addon configs

### PR DESCRIPTION
**Pull request by maciaszczykm**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
Added the short description field to addon configs.
```

### CODE REVIEW

**Comment by maciaszczykm** [comment]
Related to https://github.com/kubermatic/dashboard-v2/issues/2017.

File: api/pkg/crd/kubermatic/v1/addon_config.go [hunk 1]
```diff
@@ -18,6 +18,8 @@ type AddonConfig struct {
 
 // AddonConfigSpec specifies configuration of addon
 type AddonConfigSpec struct {
+	// ShortDescription of the configured addon
+	ShortDescription string `json:"shortDescription,omitempty"`
```
Comment (xrstf) [inline api/pkg/crd/kubermatic/v1/addon_config.go]: My Latin teacher in school used to say "A Go player plays Go. What does he do?"

Likewise documenting a field that belongs to a struct with "The field of the struct" is not very helpful. Where's the difference between description and short description? Why are there two?

**Review by xrstf — COMMENTED** [review]
Questions

File: api/pkg/crd/kubermatic/v1/addon_config.go (hunk 1 above)
Comment (maciaszczykm) [reply api/pkg/crd/kubermatic/v1/addon_config.go]: Yeah, that doesn't make sense at all, if only then to silence linters.

I've pushed updates.

**Comment by maciaszczykm** [comment]
Improvements

**Comment by xrstf** [comment]
Thanks!

/approve
/lgtm

**Pull request merged**